db/view/build_progress_virtual_reader.hh

```diff
@@ -127,8 +134,13 @@ class build_progress_virtual_reader {
                                 legacy_in_progress_row.append_cell(_legacy_generation_number_col, std::move(c));
                             }
                         });
+                        auto ck = adjust_ckey(scylla_in_progress_row.key());
+                        if (_previous_clustering_key && ck.equal(*_schema, *_previous_clustering_key)) {
```

Comment:
Is the extra _previous_clustering_key check needed? If it's empty, it won't be equal. But I guess it doesn't hurt either - so ok.

Reply:
I missed the fact that _previous_clustering_key is actually an optional. So disregard this question.

Reply:
`_previous_clustering_key` is now an `std::optional`, so I need the check before I try dereferencing it. Originally I (ab)used `clustering_key::empty()` for exactly that purpose.